torch/_inductor/fx_passes/overlap_preserving_bucketer.py

```diff
@@ -11,10 +12,58 @@
     is_reduce_scatter_tensor as is_reduce_scatter,
     is_wait_tensor,
 )
-from torch._inductor.fx_passes.overlap_scheduling import CollBucket, CollectiveInfo
+from torch._inductor.fx_passes.overlap_scheduling import CollectiveInfo
 from torch.utils._ordered_set import OrderedSet
 
 
+@dataclass
```

Comment:
Nit: might want to make slots=true here since the class looks potentially performance sensitive/ in hot paths and only modifies declared instance variables